compiler/src/iree/compiler/Codegen/Common/TransformExtensions/TransformMatchers.cpp

```diff
@@ -0,0 +1,356 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/TransformExtensions/TransformMatchers.h"
+
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+
+using namespace mlir;
+using namespace mlir::iree_compiler;
+using namespace mlir::iree_compiler::IREE;
+
+//===---------------------------------------------------------------------===//
+// StructuredOpMatcher and friends.
+//===---------------------------------------------------------------------===//
+
+bool transform_dialect::StructuredOpMatcher::match(Operation *op) {
```

Comment:
This is a cpp file, should we just using namespace `transform_dialect` or wrap them with the namespace, so we don't need `transform_dialect::` for every function implementation?

Reply:
This is following MLIR best practices as most of the code will end up in a place closer to upstream when ready.
No idea where "wrapping in the namespace" comes from but it does not follow established practice.